### PR TITLE
Add missing tests alignments for nrf54l15

### LINKED ISSUE #69216: driver: Timer / Counter initialization incorrect for timer2 with BLE on nRF52840
**Describe the bug**
An initialization value is incorrect if timer2 is enabled (and used in counter mode) with BLE (where timer0 and timer1 are disabled) on nRF52840.

Please also mention any information which could help others to understand
the problem you're facing:
 - `drivers/counter/counter_nrfx_timer.c` uses a Nordic HAL value to initialize `.timer`.
This is out-of-sync with devicetree information as timer0 and timer1 are disabled but
timer2 is enabled because:
[devicetree](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_INST)
implies that devicetree starts indexing instances from 0 based on enabled devices, and then indexes disabled devices. So, devicetree index for timer2 (0) and HAL index for timer2 (2) differ.

 - see: commit 9a27906e38343b07cbdf1b4aca5451c5886dbd41

**To Reproduce**
1. Use `samples/bluetooth/broadcaster`
2. Add following to `prj.conf `
```
CONFIG_COUNTER=y
CONFIG_NRFX_TIMER2=y
```
3. Create `boards/nrf52840dk_nrf52840.overlay` with:
```
&timer2 {
  status = "okay";
};
```
4. Build and run on nRF52840DK
5. Check value of `reg` in `init_timer`. It is 0x40008000 and should be 0x4000a000

**Expected behavior**
timer2 should operate correctly as a counter.

**Impact**
Very low priority (workaround is to revert commit above).

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: zephyr-sdk-0.16.5
 - Commit SHA or Version used: 5c455aae88d14a527bdd438adf9da4958d52c17b

### PR DESCRIPTION
**Pull request by magp-nordic**
This PR adds missing GRTC test and counter test alignment for nRF54L15. ~~Additionally, it contains fix for getting timer's instance number in the counter driver.~~

~~Fixes #69216~~ Fixed by #69966

### CODE REVIEW

File: tests/drivers/timer/nrf_grtc_timer/src/main.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	/* Absolute timeout 10ms in the future */
+	t = Z_TIMEOUT_TICKS(Z_TICK_ABS(sys_clock_tick_get() + K_MSEC(10).ticks));
+	exp_ticks = z_nrf_grtc_timer_read() + K_MSEC(10).ticks;
+	ticks = z_nrf_grtc_timer_get_ticks(t);
+	zassert_true((ticks >= exp_ticks) && (ticks <= (exp_ticks + GRTC_SLEW_TICKS)),
+		     "Unexpected result %" PRId64 " (expected: %" PRId64 ")", ticks, exp_ticks);
+}
+
+void test_main(void)
```
Comment (nordic-krch) [inline tests/drivers/timer/nrf_grtc_timer/src/main.c:40]: You need to convert it to the new ztest API.

**Review by nordic-krch — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]
Needs converting to hwmv2

**Review by aescolar — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

File: dts/bindings/timer/nordic,nrf-timer.yaml [hunk 2]
```diff
@@ -32,3 +32,15 @@ properties:
...
+    type: string
+    required: true
+    description: |
+      The TIMER instance id as used in the MDK and documentation. For example for TIMER2:
+
+        instance = "2";
+
+      And for TIMER00:
+
+        instance = "00";
```
Comment (henrikbrixandersen) [inline dts/bindings/timer/nordic,nrf-timer.yaml:46]: We have previously rejected similar work-arounds for other SoCs. Could we find another way around this?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: dts/bindings/timer/nordic,nrf-timer.yaml (hunk 2 above)
Comment (gmarull) [reply dts/bindings/timer/nordic,nrf-timer.yaml:46]: I won't nack here, but I agree that this is a DT usage violation we have rejected multiple times. If HALs do not match DT expectations, they should be modified to do so, use other layers (eg `nrf_`), or not use them at all.

Comment (magp-nordic) [reply dts/bindings/timer/nordic,nrf-timer.yaml:46]: This change is part of one of the commits fixing https://github.com/zephyrproject-rtos/zephyr/issues/69216 . Here is an alternative fix for the same issue: #69966. If it gets merged, I will drop these changes from my PR.

**Review by aescolar — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
I see we only have custom timer driver tests for nRF timers. Wouldn't it possible to have a generic timer test covering the standard Zephyr timer driver API?

**Comment by carlescufi** [comment]
> I see we only have custom timer driver tests for nRF timers. Wouldn't it possible to have a generic timer test covering the standard Zephyr timer driver API?

@nordic-krch any comments here?

**Pull request merged**